hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/StoreFileListFile.java

```diff
@@ -86,6 +86,8 @@ class StoreFileListFile {
 
   static final Pattern TRACK_FILE_PATTERN = Pattern.compile("^f(1|2)\\.\\d+$");
```

Comment:
What about changing this regex to make \\d+ optional? Like `^f(1|2)\\(.\\d+)*$`? And then we use the Matcher's group method to extract the timestamp, if it does not present, we just set it to 0 or -1, so it will placed at the last when comparing with the files written by newer version.
In this way we do not need to rename the file since rename coud fail and we also need to deal with it...

Reply:
I agree with your suggestion that it's better to handle missing timestamps by setting them to 0, rather than dealing with the potential issues of renaming.

https://github.com/apache/hbase/pull/6552/commits/f2d54bb4ecf07330a47bcb142767ac6c32cc39aa